examples/compiler/01_lowering.ipynb

```diff
@@ -0,0 +1,352 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Lowering\n",
+    "\n",
+    "In the first compilation pass, three main tasks are carried out: _indexification_, _substitution_, and _domain-alignment_.\n",
+    "\n",
+    "\n",
+    "### Indexification\n",
+    "\n",
+    "The _indexification_ consists of converting functions into actual arrays of accesses. An access always keeps a reference to its originating function. For instance, all accesses to `u` such as `u[t, x + 1]` and `u[t + 1, x − 2]`would store a pointer to the same, user-defined `Function` _u(t, x)_. This metadata is exploited throughout the various compilation passes. \n",
+    "\n",
+    "To see this, let's create a `TimeFunction` named `u`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "u(time, x)"
+      ]
+     },
+     "execution_count": 1,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from devito import Grid, TimeFunction\n",
+    "u = TimeFunction(name='u', grid=Grid((3,)), save=1)\n",
+    "u"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note that `u` has one temporal dimension and one spacial dimension, that we will call `t` and `x` here, respectively."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "(time, x)"
+      ]
+     },
+     "execution_count": 2,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "t,x = u.dimensions\n",
+    "t,x"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Indexifying functions allows us to write finite differences stencils from it. Functions (of type `Function`, `TimeFunction` and `SparceFunction`) can be indexified by the method `.indexify()`, as illustrated bellow."
```

Comment:
Sparse

Reply:
I'd entirely drop "Indexifying functions allows us to write finite differences stencils from it."

and then:

"can be indexified in two equivalent ways, either via the `.indexify()` method [and then you show the notebook cell] or via explicit indexing, as typically done in user code [and then you add a new cell showing that ``f[time, x] -> f[time, x]``]"

Reply:
thank you george

@FabioLuporini fixed partially. I didn't get what you mean by "or via explicit indexing, as typically done in user code [and then you add a new cell showing that f[time, x] -> f[time, x]]"